.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,161 @@
+# documentation: https://help.github.com/en/articles/workflow-syntax-for-github-actions
+name: EasyBuild framework unit tests
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python: [2.7, 3.5, 3.6, 3.7]
+        modules_tool: [Lmod-6.6.3, Lmod-7.8.22, Lmod-8.1.14, modules-tcl-1.147, modules-3.2.10, modules-4.1.4]
+        module_syntax: [Lua, Tcl]
+        # exclude some configuration for non-Lmod modules tool:
+        # - don't test with Lua module syntax (only supported in Lmod)
+        # - don't test with Python 3.5 and 3.7 (2.7 and 3.7), to limit test configurations
+        exclude:
+          - modules_tool: modules-tcl-1.147
+            module_syntax: Lua
+          - modules_tool: modules-3.2.10
+            module_syntax: Lua
+          - modules_tool: modules-4.1.4
+            module_syntax: Lua
+          - modules_tool: modules-tcl-1.147
+            python: 3.5
+          - modules_tool: modules-tcl-1.147
+            python: 3.7
+          - modules_tool: modules-3.2.10
+            python: 3.5
+          - modules_tool: modules-3.2.10
+            python: 3.7
+          - modules_tool: modules-4.1.4
+            python: 3.5
+          - modules_tool: modules-4.1.4
+            python: 3.7
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{matrix.python}}
+        architecture: x64
+
+    - name: install OS & Python packages
+      run: |
+        # for modules tool
+        sudo apt-get install lua5.1 liblua5.1-0-dev lua-filesystem lua-posix lua-filesystem tcl tcl-dev
```

Comment:
Should probably use lua5.2 instead but not important.

Reply:
Well, I can update it to `lua5.2`, but we're not testing Lmod here... ;)

Why would you bump it to Lua 5.2?

Reply:
Mostly because 5.1 is old, and 5.3 has been out for quite a while... and Lmod supports 5.2
So as long as it's running on Ubuntu18.04 5.2 is a more "natural" choice.

But as I said, not important, just a suggestion.

Reply:
Well, ok, doesn't hurt to bump to 5.2